evaluate/report/collection.go

```diff
@@ -86,10 +86,13 @@ func (a AssessmentPerModelPerLanguagePerRepository) CollapseByModel() Assessment
 }
 
 // WalkByScore walks the given assessment metrics by their score.
-func WalkByScore(assessmentsPerModel map[string]metrics.Assessments, function func(model string, assessment metrics.Assessments, score uint) error) error {
+func WalkByScore(assessmentsPerModel AssessmentPerModel, function func(model model.Model, assessment metrics.Assessments, score uint) error) error {
```

Comment:
How about making `WalkByScore` a method of `AssessmentPerModel`?

Reply:
Sounds good, will change